packages/node_modules/@webex/plugin-meetings/src/reachability/index.js

```diff
@@ -0,0 +1,283 @@
+    const localSDPList = [];
+    const peerConnectionMap = {};
+
+    return new Promise(((resolve, reject) => {
+      clusterList.forEach((cluster) => {
+        const clusterId = cluster;
+
+        peerConnectionManager.createOffer({receiveAudio: true, receiveVideo: false})
+          .then((peerConnection) => {
+            peerConnectionMap[clusterId] = peerConnection;
+
+            const localSDP = {
+              sdp: peerConnection.sdp,
+              clusterId
+            };
+
+            localSDPList.push(localSDP);
+
+            if (localSDPList.length === clusterList.length) {
+              localSDPData.peerConnection = peerConnectionMap;
+              localSDPData.localSDPList = localSDPList;
+              resolve(localSDPData);
+            }
+          })
+          .catch((error) => {
+            reject(new Error(`Error in getLocalSDP : ${error}`));
+            console.error(`Error in getLocalSDP : ${error}`);
+          });
+      });
+    }));
+  };
+
+  /**
+   * calculates time to establish connection
+   * @param {object} iceResult iceResults
+   * @returns {object} result
+   */
+  parseIceResultToReachabilityResult = (iceResult) => {
+    if (!iceResult || !Object.keys(iceResult).length) {
+      return {};
+    }
+    const result = {};
+    for (const clusterId of Object.keys(iceResult)) {
+      let reachability = {};
+      if (iceResult[clusterId].start > 0 && iceResult[clusterId].stop > 0) {
+        const latency = iceResult[clusterId].stop - iceResult[clusterId].start;
+        reachability = {
+          reachable: true,
+          latencyInMilliseconds: latency
+        };
+      }
+      else {
+        reachability = {reachable: false};
+      }
+      result[clusterId] = {udp: reachability};
+    }
+    return result;
+  };
+
+  /**
+   * cleans all the peerconnection
+   * @param {object} peerConnectionMap peerconnectionList
+   * @returns {object} localSDPData
+   */
+  cleanUpPeerConnections = (peerConnectionMap) => {
+    for (const key of Object.keys(peerConnectionMap)) {
+      const pc = peerConnectionMap[key];
+      if (pc) {
+        console.info(`Cleaning up peer connection created for cluster : ${key}`);
+        pc.close();
+      }
+    }
+  };
+
+  /**
+   * sets the success callback for peerconnection
+   * @param {object} pc peerconnection
+   * @returns {object} iceState
+   */
+  setIceCallBackHandlers = (pc) => {
+    let timeout = null;
+    const iceStats = {start: 0, stop: 0};
+    return new Promise(((resolve) => {
+      function iceComplete(isSuccessful, timeout, iceStats) {
+        if (isSuccessful) {
+          if (timeout) {
+            window.clearTimeout(timeout);
+            timeout = undefined;
+          }
+        }
+        pc.close();
+        resolve({iceStats});
+      }
+
+      timeout = window.setTimeout(() => {
+        console.info(`${this.clusterId} : onTimer, ice timeout!`);
+        iceComplete(false, timeout, iceStats);
+      }, 5000);
```

Comment:
Why do we need a timeout here?

Reply:
If for some reason the ice candidate fails we have a timeout . Its just going to resolve with a higher time thats it . 

By default the reachability is not going to wait for anything it just does the processing and populate the local storage